### ISSUE TITLE
Current working directory status information stays same after a new login to another zone

### ISSUE THREAD

**Issue by mstfdkmn** [irods/irods#7789]
Even after I login to another zone, my current working directory keeps having the information of my former session at previous zone. However the irods_environment.json file is updated.

current zone info
```
root@33d6bc1286de:/# ienv | grep irods_cwd
irods_cwd - /icts/home/operator
```

login to another zone
```
root@33d6bc1286de:/# imiscsvrinfo | grep rodsZone
rodsZone=ghum

root@33d6bc1286de:/# ienv | grep irods_cwd
irods_cwd - /icts/home/operator
root@33d6bc1286de:/# ipwd
/icts/home/operator

```
And only `ils` command return an errors explaining it.

```
root@33d6bc1286de:/# ils
remote addresses: 134.58.67.238 ERROR: lsUtil: srcPath /icts/home/operator does not exist or user lacks access permission
```
Why does not the new login update it? Why do queries stick to old state?

**Comment by korydraughn** [comment]
It's not clear to me what you mean by **login to another zone**.

Please post what you did exactly.

**Comment by alanking** [comment]
I believe you need to run `iexit` to clear out the file(s) tracking current working directories. Have you tried running `iexit` before running `iinit`?

**Comment by korydraughn** [comment]
To add to what @alanking said, there's also the `~/.irods/irods_environment.json.<terminal_pid>` file. When using `icd`, the current working collection is tied to the terminal being used.

Overall, I second @alanking's suggestion. `iinit` is just a tool for generating the environment file when it doesn't exist.

**Comment by trel** [comment]
`iinit` also saves a new `.irodsA` file.

**Comment by mstfdkmn** [comment]
Yes, after running `iexit` if a new session is made to another zone, all works normal.

**Comment by alanking** [comment]
Okay, good. Perhaps this is something we need to include in our documentation somewhere, then?

**Comment by trel** [comment]
`iinit` *could* look and compare the cwd path in the existing/active `irods_environment.json.<terminal_pid>` file and 'reset' it if the zone information doesn't match (maybe by just deleting the file).

**Comment by korydraughn** [comment]
Hmm. What happens if the user has multiple terminals open with different CWDs captured across several `irods_environment.json.<terminal_pid>` files and then proceeds to run `iinit`?

**Comment by trel** [comment]
The CWDs are per-terminal (per-pid) already.

Changing `iiniit` would only affect the current terminal's/pid's CWD.

Does that answer your question?

**Comment by korydraughn** [comment]
Yep. I don't sense any obvious drawbacks to your solution.

Removing the file seems like the safest approach. Should `iinit` or `iexit` be responsible for that?

**Comment by trel** [comment]
it's not `iexit`'s job - that valid CWD could be used on the next `iinit`.

`iinit` should only remove it if the new zone doesn't match the existing/old CWD's zone.

**Comment by korydraughn** [comment]
> `iinit` should only remove it if the new zone doesn't match the existing/old CWD's zone.

Here's another question.

Let's say a user authenticates as `bobby` against tempZone and uses `icd` to change to a different collection. The user then proceeds to modify their environment file and changes the `irods_user_name` property to `alice`. The user then uses `iinit` and successfully authenticates as `alice`.

The `alice` iRODS user does not have permissions on the collection captured in CWD. The user forgets they are sitting in a collection they don't have access to and executes a command which results in an error. Keep in mind, the user did not switch zones.

What are your thoughts on this? Is this a user issue? Is this out-of-scope?

**Comment by trel** [comment]
I think that's the same scenario when `alice` is logged in and just goes and edits the existing `irods_environment.json.<terminal_pid>` file to have a collection path that she does not have access to.

`ils` will then fail - no permission to that collection.  Any error message should show the full logical path, and then the user would 'see' where they are and 'fix' it...

`icd` will either update the `irods_environment.json.<terminal_pid>` file to a correct value again, or... if used alone, remove the `irods_environment.json.<terminal_pid>` file altogether.

**Comment by korydraughn** [comment]
That's fair.

**Comment by alanking** [comment]
Okay, so... we have an enhancement for `iinit` and possibly a documentation exercise around `iexit`?

**Comment by trel** [comment]
yes for the enhancement for `iinit`.

what is the documentation for `iexit`?  that it deletes the .pid file and therefore, its remembering of any cwd?

**Comment by alanking** [comment]
The `iexit` help text already has this information, so I suppose it would be more a place pointing people to `iexit` in this situation. I guess it wouldn't be that helpful. So, maybe just an enhancement for `iinit`.

**Comment by luijs** [comment]
There curently also is an undocumented option of adding a line to your environment file allowing for multiple zone usage at the same time without re-using `iinit`:
`"irods_authentication_file": "/home/myuser/.irods/.irodsA_yoda",`
We combine this with mutiple irods_environment.json_zonename files and `cp` commands to switch between environments. Could an `iinit --environment=yoda` sort of option help here as well?

**Comment by trel** [comment]
The string 'yoda' would have to be mapped somehow to a particular location.   Where would that live?

Where would you expect to see the `irods_authentication_file` option documented, that it is not today?

**Comment by luijs** [comment]
True, the string yoda is a bit too cryptic. It probably would have to be a filename instead, like irods_environment_yoda.json that also lives in ~/.irods.

The documentation could be here:

https://docs.irods.org/4.3.2/system_overview/configuration/#irodsirods_environmentjson
